app/src/main/java/com/lucas/medicaltools/service/MedicalToolsInteractor.kt

```diff
@@ -0,0 +1,48 @@
+package com.lucas.medicaltools.service
+
+import com.lucas.medical_equip.repository.MedicalTool
+import com.lucas.medical_equip.service.RetrofitBuilder
+import com.lucas.medicaltools.MainViewState
+import com.lucas.medicaltools.repository.MedicalToolsRepository
+import io.reactivex.Observable
+import io.reactivex.schedulers.Schedulers
+import retrofit2.Response
+
+object MedicalToolsInteractor {
+    lateinit var medicalToolCache: Response<List<MedicalTool>>
+    fun getMedicalTools(filter: String): Observable<MainViewState>? {
+        if (filter.isNullOrBlank()) {
```

Comment:
repository.getCache().flatMap{ if filter is null or blank ? it (the response): filter(it)}